File: packages/js-sdk/src/regex/detectors/nl/bsn.ts
// Dutch BSN (Burgerservicenummer) detector

import { RegexDetector } from '../../base'
import { EntityType } from '../../entities'
import { registerRegion } from '../../registry'
import { nlBsn11test } from '../../validators'

class NlBsnDetector extends RegexDetector {
  entityType = EntityType.NL_BSN
  score = 0.85
  contextKeywords = [
    'bsn', 'burgerservicenummer', 'sofi', 'citizen service',
  ]
  contextRequired = true

  pattern = /\b\d{9}\b/g
  validator = nlBsn11test
}

registerRegion('nl', NlBsnDetector)

export { NlBsnDetector }

File: packages/python-sdk/src/blindfold/regex/detectors/nl/bsn.py
"""Dutch BSN (Burgerservicenummer) detector"""

import re

from ...base import RegexDetector
from ...entities import EntityType
from ...registry import register_region
from ...validators import nl_bsn_11test


@register_region("nl")
class BsnDetector(RegexDetector):
    entity_type = EntityType.NL_BSN
    score = 0.85
    context_keywords = [
        "bsn", "burgerservicenummer", "sofi", "citizen service",
    ]
    context_required = True
    context_window = 50

    # 9 digits
    pattern = re.compile(
        r"\b\d{9}\b"
    )
    validator = staticmethod(nl_bsn_11test)
